tests/unit/tasks/models/tasksManager.spec.ts

```diff
@@ -440,12 +441,67 @@ describe('TasksManager', () => {
       });
 
       mockFindTasks.mockResolvedValue([exportTaskMock]);
-      //mockFindTasks.mockResolvedValueOnce([finalizeTaskMock]);
       mockGetJob.mockResolvedValue(exportJobMock);
       // action
       await tasksManager.handleTaskNotification(exportTaskMock.id);
       // expectation
       expect(mockUpdateJob).not.toHaveBeenCalled();
     });
+
+    it('Should fail a job on a failed seeding task', async () => {
+      // mocks
+      const { tasksManager, mockFindTasks, mockUpdateJob, jobDefinitionsConfigMock, mockGetJob } = testContext;
+      const seedingJob = getSeedingJobMock();
+      const seedTaskMock = getTaskMock(seedingJob.id, {
+        type: jobDefinitionsConfigMock.tasks.seed,
+        status: OperationStatus.FAILED,
+        reason: 'some error reason',
+      });
+
+      mockFindTasks.mockResolvedValueOnce([seedTaskMock]);
+      mockGetJob.mockResolvedValue(seedingJob);
+      // action
+      await tasksManager.handleTaskNotification(seedTaskMock.id);
+      // expectation
+      expect(mockUpdateJob).toHaveBeenCalledWith(seedingJob.id, { status: OperationStatus.FAILED, reason: 'some error reason' });
+    });
+
+    it('Should update job on a completed seeding task', async () => {
+      const { tasksManager, mockGetJob, mockFindTasks, jobDefinitionsConfigMock, mockCreateTaskForJob, mockUpdateJob } = testContext;
+      const seedingJob = getSeedingJobMock({ taskCount: 6, completedTasks: 4 });
+      const seedTaskMock = getTaskMock(seedingJob.id, { type: jobDefinitionsConfigMock.tasks.seed, status: OperationStatus.COMPLETED });
+      const desiredPercentage = calculateTaskPercentage(seedingJob.completedTasks, seedingJob.taskCount);
+
+      mockFindTasks.mockResolvedValue([seedTaskMock]);
+      mockGetJob.mockResolvedValue(seedingJob);
+      // action
+      await tasksManager.handleTaskNotification(seedTaskMock.id);
+      // expectation
+      expect(mockFindTasks).toHaveBeenCalledTimes(1);
+      expect(mockGetJob).toHaveBeenCalledTimes(1);
+      expect(mockCreateTaskForJob).not.toHaveBeenCalled();
```

Comment:
why are you explicitly check for it not being called?

Reply:
removed